admin/services/data-access/group.data.service.js

```diff
@@ -65,8 +139,158 @@ groupDataService.update = async function (id, data) {
   })
 }
 
+/**
+ * Update group.
+ * @param id
+ * @param name
+ * @returns {Promise}
+ */
+groupDataService.sqlUpdate = async (id, name) => {
+  const params = [
+    {
+      name: 'id',
+      value: id,
+      type: TYPES.Int
+    },
+    {
+      name: 'name',
+      value: name,
+      type: TYPES.NVarChar
+    },
+    {
+      name: 'updatedAt',
+      value: moment.utc(),
+      type: TYPES.DateTimeOffset
+    }
+  ]
+  return sqlService.modify(
+    `UPDATE ${sqlService.adminSchema}.[group] 
+    SET name=@name, updatedAt=@updatedAt 
+    WHERE [id]=@id`,
+    params)
+}
+
+/**
+ * Get pupils per group.
+ * @returns {Promise<*>}
+ */
+groupDataService.getPupilsPerGroup = async () => {
+  const sql = `SELECT group_id, COUNT (pupil_id) AS total_pupils 
+    FROM ${sqlService.adminSchema}.[pupilGroup] 
+    GROUP BY group_id`
+  return sqlService.query(sql)
+}
+
+/**
+ * Update pupils assigned to group.
+ * @param groupId
+ * @param pupilIds
+ * @returns {Promise<void>}
+ */
+groupDataService.sqlAssignPupilsToGroup = async (groupId, pupilIds) => {
+  // First, delete all pupils for the selected groupId
+  const sqlDelete = `DELETE FROM ${sqlService.adminSchema}.[pupilGroup] 
+    WHERE group_id=@groupId`
+
+  const paramsDel = [
+    {
+      name: 'groupId',
+      value: groupId,
+      type: TYPES.Int
+    }
+  ]
+  await sqlService.modify(sqlDelete, paramsDel)
+
+  // Next, insert passed pupils
+  if (pupilIds.length > 0) {
+    let sqlInsert = ''
+    let paramsInsert
+
+    pupilIds.forEach(async p => {
+      if (p) {
+        paramsInsert = [
+          {
+            name: 'groupId',
+            value: groupId,
+            type: TYPES.Int
+          },
+          {
+            name: 'pupilId',
+            value: p,
+            type: TYPES.Int
+          }
+        ]
+
+        // @TODO: This can be improved to avoid the loop with a syntax as follows:
+        // INSERT INTO #SQLAuthority (ID, Value)
+        // VALUES (1, 'First'), (2, 'Second'), (3, 'Third');
+        sqlInsert = `INSERT INTO ${sqlService.adminSchema}.[pupilGroup] (group_id, pupil_id) VALUES (@groupId, @pupilId)`
```

Comment:
This loop needs to be refactored before merge please.  Not a TODO item.

It can be written in a single SQL statement fairly trivially:

`INSERT INTO ${sqlService.adminSchema}.[pupilGroup] 
(group_id, pupil_id) 
VALUES 
(42, 1),
(42, 2),
(42, 3),
...
`

Reply:
@GeorgeChatzigiannis can you please apply this same concept to your loop block that contains the multiple insert statements